reshape.hhs

```diff
@@ -0,0 +1,107 @@
+/**
+ * @author Jianan Lin (林家南)
+ * @param1 input - an array / matrix / tensor
+ * @param2 sizes - an array such as [2, 3, 2] or [2, -1]
+ * @returns - an 1-d array with specified size
+ * 
+ */
+
+
+function reshape(input, sizes = [-1]) {
+
+    *import math: flatten
+    *import math: ndim
+    *import math: deep_copy
+    *import math: is_number
+
+    // wrong argument number
+    if (arguments.length === 0) {
+        throw new Error('Exception occurred in reshape - no argument given');
+    }
+    else if (length > 2) {
```

Comment:
i might just be tired but im not sure why this is working when `length` isn't declared anywhere... should crash before runtime but just to be safe: i assume you meant
```suggestion
    else if (arguments.length > 2) {
```

other than that looks great. Good work